jina/serve/runtimes/gateway/request_handling.py

```diff
@@ -61,17 +62,35 @@ def __init__(
                 labelnames=('runtime_name',),
             ).labels(runtime_name)
 
+            self._failed_requests_metrics = Gauge(
+                'number_of_failed_requests',
+                'Number of failed requests',
+                registry=metrics_registry,
+                namespace='jina',
+                labelnames=('runtime_name',),
+            ).labels(runtime_name)
+
+            self._successful_requests_metrics = Gauge(
+                'number_of_successful_requests',
+                'Number of successful requests',
+                registry=metrics_registry,
+                namespace='jina',
+                labelnames=('runtime_name',),
+            ).labels(runtime_name)
+
         else:
             self._receiving_request_metrics = None
             self._pending_requests_metrics = None
+            self._failed_requests_metrics = None
+            self._successful_requests_metrics = None
 
     def _update_start_request_metrics(self, request: 'Request'):
         if self._receiving_request_metrics:
             self._request_init_time[request.request_id] = time.time()
         if self._pending_requests_metrics:
             self._pending_requests_metrics.inc()
 
-    def _update_end_request_metrics(self, result: 'Request'):
+    def _update_end_request_metrics(self, result: 'Request', e: Exception = None):
```

Comment:
Can we rename e to `excpetion` or something a bit more verbose ? I find it hard to read the code otherwise